tests/suites/os/tests/secureboot/index.js

```diff
@@ -1,145 +1,193 @@
-					['ls', securebootEfiVarPath, '&>/dev/null',
-						'&&', 'echo', 'true',
-						'||', 'echo', 'false'
-					],
-					this.link,
-				).then(output => { return output === 'true'});
-
-				// https://wiki.archlinux.org/title/Unified_Extensible_Firmware_Interface/Secure_Boot#Checking_Secure_Boot_status
-				const securebootEnabled = await this.worker.executeCommandInHostOS(
-						[
-							'od', '--address-radix=n', '--format=u1', securebootEfiVarPath,
-								'|', 'tr', '-s', "' '",
-								'|', 'cut', "-d' '", '-f6'
-						],
-						this.link,
-				).then(output => output === '1' );
-
-				// Used after tampering with verified boot files
-				const resetWorker = async () => {
-					await this.worker.off();
-					await this.worker.flash(this.os.image.path);
-					await this.worker.on();
-
-					await this.worker.addSSHKey(this.sshKeyPath);
-				}
-
-				const waitForSerialOutput = async(pattern, slice=0, retries=60, delay=1000) => this.utils.waitUntil(
-					async () => this.worker.fetchSerial().then(
-						serialLogs => serialLogs.split('\n')
-																		.slice(slice)
-																		.join('\n')
-																		.match(pattern)
-					), false, retries, delay
-				);
-
-				const testFullDiskEncryption = async () => Promise.all(
-					[
-						{ pattern: '/^\\/mnt\\/boot$/', name: 'Boot partition' },
-						{ pattern: '/^\\/mnt\\/sysroot\\/active$/', name: 'Root partition' },
-						{ pattern: '/^\\/mnt\\/state$/', name: 'State partition' },
-						{ pattern: '/^\\/mnt\\/data$/', name: 'Data partition' },
-					].map(args => test.resolveMatch(
-							this.worker.executeCommandInHostOS(
-								[
-									'lsblk', '-nlo', 'MOUNTPOINT,TYPE',
-									'|', 'awk', `'$1 ~ ${args.pattern} { print $2 }'`
-								],
-								this.link,
-							),
-							/crypt/,
-							`${args.name} is encrypted`,
-						)
-					)
-				);
+class secureBoot {
+	constructor(test, worker, context, imagePath) {
+		this.test = test;
+		this.worker = worker;
+		this.link = context.get().link;
+		this.imagePath = imagePath;
+		this.sshKeyPath = context.get().sshKeyPath;
+		this.utils = context.get().utils;
+	}
 
-				const testModuleVerification = async() => {
-					await test.resolves(
-						this.worker.executeCommandInHostOS('modprobe pcan_netdev', this.link),
-						'Module with valid signature loads',
-					);
+	async resetDUT() {
+		await this.worker.off();
+		await this.worker.flash(this.imagePath);
+		await this.worker.on();
 
-					await this.worker.pushContainerToDUT(
-						this.link,
-						`${__dirname}/kernel-module-build`,
-						'load',
-					);
+		await this.worker.addSSHKey(this.sshKeyPath);
+	}
 
-					return test.resolveMatch(
+	async waitForSerialOutput(pattern, slice=0, retries=60, delay=1000) {
+		await this.utils.waitUntil(
+			async () => this.worker.fetchSerial().then(
+				serialLogs => serialLogs.split('\n')
+				.slice(slice)
+				.join('\n')
+				.match(pattern)
+			), false, retries, delay
+		);
+	}
```

Comment:
```suggestion
				(serialLogs) => {
				  serialLogs.split('\n')
				  .slice(slice)
				  .join('\n')
				  .match(pattern)
				}
			), false, retries, delay
		);
	}
```

NIT: Improving readability